// @org/db-models/dist/entities/domain.entity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainEntity = exports.deploymentTypeMap = exports.DeploymentType = exports.domainStatusMap = exports.DomainStatus = void 0;
var DomainStatus;
(function (DomainStatus) {
    DomainStatus["BAN"] = "BAN";
    DomainStatus["FREEZE"] = "FREEZE";
    DomainStatus["DO_NOT_WORK"] = "DO_NOT_WORK";
    DomainStatus["REPLACEMENT"] = "REPLACEMENT";
    DomainStatus["ACTIVE"] = "ACTIVE";
    DomainStatus["FAILED"] = "FAILED";
    DomainStatus["QUEUE"] = "QUEUE";
    DomainStatus["INACTIVE"] = "INACTIVE";
})(DomainStatus || (exports.DomainStatus = DomainStatus = {}));
exports.domainStatusMap = {
    BAN: DomainStatus.BAN,
    FREEZE: DomainStatus.FREEZE,
    DO_NOT_WORK: DomainStatus.DO_NOT_WORK,
    REPLACEMENT: DomainStatus.REPLACEMENT,
    ACTIVE: DomainStatus.ACTIVE,
    FAILED: DomainStatus.FAILED,
    QUEUE: DomainStatus.QUEUE,
    INACTIVE: DomainStatus.INACTIVE,
};
var DeploymentType;
(function (DeploymentType) {
    DeploymentType["FORGE"] = "FORGE";
    DeploymentType["SSH"] = "SSH";
})(DeploymentType || (exports.DeploymentType = DeploymentType = {}));
exports.deploymentTypeMap = {
    FORGE: DeploymentType.FORGE,
    SSH: DeploymentType.SSH,
};
class DomainEntity {
    id;
    name;
    serverId;
    repoId;
    sshUrl;
    price;
    hasSSL;
    campaign;
    type;
    status;
    deploymentType;
    constructor(partial) {
        Object.assign(this, partial);
    }
}
exports.DomainEntity = DomainEntity;
